# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/StencilXdr.cpp
# Commit: 8af30c6d9bb9
# Full Hash: 8af30c6d9bb91a1ed9a5bee712252bbe1bd20d85
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2020-11-14 09:46:25
# Regressor Bug: 1675241
# File Overlap Count: 1
# Description:
#   Bug 1675241 - Part 11: Use TaggedParserAtomIndex in ScriptThingVariant. r=tcampbell
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D96918
# ==============================================================================

diff -r cb0e58368cd4 -r 8af30c6d9bb9 js/src/frontend/StencilXdr.cpp
--- a/js/src/frontend/StencilXdr.cpp	Sat Nov 14 00:03:28 2020 +0000
+++ b/js/src/frontend/StencilXdr.cpp	Sat Nov 14 00:03:36 2020 +0000
@@ -31,7 +31,7 @@
 static XDRResult XDRScriptThingVariant(XDRState<mode>* xdr,
                                        ScriptThingVariant& thing) {
   enum class ScriptThingKind {
-    ScriptAtom,            // JSAtom*
+    ScriptAtom,            // Index.
     NullScriptThing,       // nothing.
     BigIntIndex,           // Index.
     ObjLiteralIndex,       // Index.
@@ -44,7 +44,7 @@
   uint32_t index = UINT32_MAX;
 
   struct KindMatcher {
-    ScriptThingKind operator()(ScriptAtom& atom) {
+    ScriptThingKind operator()(TaggedParserAtomIndex& index) {
       return ScriptThingKind::ScriptAtom;
     }
     ScriptThingKind operator()(NullScriptThing&) {
@@ -77,13 +77,13 @@
 
   MOZ_TRY(xdr->codeEnum32(&kind));
 
-  const ParserAtom* atom = nullptr;
+  TaggedParserAtomIndex atom;
   if (kind == ScriptThingKind::ScriptAtom) {
     MOZ_ASSERT(index == UINT32_MAX);
     if (mode == XDR_ENCODE) {
-      atom = thing.as<ScriptAtom>();
+      atom = thing.as<TaggedParserAtomIndex>();
     }
-    MOZ_TRY(XDRParserAtom(xdr, &atom));
+    MOZ_TRY(XDRTaggedParserAtomIndex(xdr, &atom));
   } else if (kind == ScriptThingKind::BigIntIndex ||
              kind == ScriptThingKind::ObjLiteralIndex ||
              kind == ScriptThingKind::RegExpIndex ||
@@ -98,9 +98,9 @@
   if (mode == XDR_DECODE) {
     switch (kind) {
       case ScriptThingKind::ScriptAtom:
-        // `atom` is initialized above if `kind` == ScriptAtom
-        MOZ_ASSERT(atom != nullptr);
-        thing.emplace<ScriptAtom>(atom);
+        // `atom` is initialized above if `kind` == TaggedParserAtomIndex
+        MOZ_ASSERT(atom);
+        thing.emplace<TaggedParserAtomIndex>(atom);
         break;
       case ScriptThingKind::NullScriptThing:
         thing.emplace<NullScriptThing>();
